.github/workflows/pull-request.yml

```
@@ -10,6 +10,11 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 20
+        registry-url: https://registry.npmjs.org/
```

Bump this to v4 of the action? Shouldn't need to specify the registry, as npmjs is the default?  Setting `cache: 'npm'` should also speed up the `npm install` step by using a cache between builds.

```suggestion
    - uses: actions/setup-node@v4
      with:
        node-version: 20
        cache: 'npm'
```

---

good idea, adding it in now